roles/telemetry_logging/tasks/logging_pod_tests.yml

```diff
@@ -0,0 +1,32 @@
+---
+
+- name: Get Pod Instance "{{ pod_status_str }}" 
+  ansible.builtin.shell:
+    cmd: |
+      oc get pods -n  "{{ nspace }}" | grep "{{ item }}" | grep "{{ pod_status_str }}" | awk '{print $1;}'
+  register: podinstance
+  changed_when: false
+  failed_when: false
+
+- name: Check terminated pod 
+  ansible.builtin.shell:
+    cmd: oc get pod -n "{{ nspace }} {{ podinstance.stdout }}"
+  register: output
+  changed_when: false
+  failed_when: false
+
+- name: Test pod does not exist - "{{ item }}"
+  ansible.builtin.set_fact:
+     exists: FAILED
+
+- name: Test pod exists - "{{ item }}"
+  ansible.builtin.set_fact:
+    exists: PASSED
+  when: 
+    - output.rc == 0
+    - podinstance.stdout != ""
+
```

Comment:
I have a suggestion for improving the efficiency here.

if you use the ``failed_when`` option for the "Get pod instance" task, you can then set the var ``exists`` based on the status of that test.

The success filter should be useful here.
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/success_test.html
you can do something like:

```
- set_fact:
     exists: "{{ output is ansible.builtin.success }}"
```